pkgs/top-level/all-packages.nix

```diff
@@ -23887,6 +23887,17 @@ with pkgs;
       };
     } ../os-specific/darwin/darwin-min-version-hook/setup-hook.sh;
 
+  ### DEVELOPMENT / TESTING TOOLS
+
+  atf = callPackage ../by-name/at/atf/package.nix {
+    stdenv =
+      # atf is a dependency of libiconv. Avoid an infinite recursion with `pkgsStatic` by using a bootstrap stdenv.
+      if stdenv.hostPlatform.isDarwin then
+        darwin.bootstrapStdenv
+      else
+        stdenv;
```

Comment:
All a bootstrap stdenv does is not propagate any packages. ATF can’t use a regular stdenv on Darwin because otherwise the SDK in the stdenv would propagate libiconv, which depends on ATF for tests.